server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/cs/factory/DeployedCacheStoreFactory.java

```diff
@@ -18,14 +17,22 @@
  */
 public class DeployedCacheStoreFactory implements CacheStoreFactory {
 
-   private Map<String, DeployedCacheStoreMetadata> deployedCacheStores = Collections.synchronizedMap(new HashMap<String, DeployedCacheStoreMetadata>());
+   private static final int TIMEOUT = 20;
+   private static final TimeUnit TIMEOUT_UNITS = TimeUnit.SECONDS;
```

Comment:
To be honest - I'm not sure. This issue happens only on very slow environments, so I would say it is OK to hardcode it (maybe I would even increase it to 1 minute). But if we decided to make it configurable - where should we place the configuration?